### PR TITLE
ID3v2: Create more concrete types for `FrameValue` variants

### PR DESCRIPTION
**Pull request by Serial-ATA**
This makes working with `FrameValue` nicer, as there are no longer struct variants and reused types.

`FrameValue` before:

```rs
pub enum FrameValue {
	Comment(LanguageFrame),
	UnSyncText(LanguageFrame),
	Text {
		/// The encoding of the text
		encoding: TextEncoding,
		/// The text itself
		value: String,
	},
	UserText(EncodedTextFrame),
	URL(String),
	UserURL(EncodedTextFrame),
	Picture {
		/// The encoding of the description
		encoding: TextEncoding,
		/// The picture itself
		picture: Picture,
	},
	Popularimeter(Popularimeter),
	Binary(Vec<u8>),
	UniqueFileIdentifier(UniqueFileIdentifierFrame),
}
```

After:

```rs
pub enum FrameValue {
	Comment(CommentFrame),
	UnSyncText(UnsynchronizedTextFrame),
	Text(TextInformationFrame),
	UserText(ExtendedTextFrame),
	URL(UrlLinkFrame),
	UserURL(ExtendedUrlFrame),
	Picture(AttachedPictureFrame),
	Popularimeter(Popularimeter),
	Binary(Vec<u8>),
	UniqueFileIdentifier(UniqueFileIdentifierFrame),
}
```

It also allowed for all of the parsing logic to be moved to the respective structs, making things cleaner.

### CODE REVIEW

**Comment by uklotzde** [comment]
Having distinct types for the variants is more flexible and avoids redundant and boilerplate type conversion.

**Review by uklotzde — CHANGES_REQUESTED** [review]
I would prefer to follow consistent naming rules for `FrameValue::UnSyncText` like for all other frame value types, i.e. `FrameValue::UnsynchronizedText(UnsynchronizedTextFrame)`.

**Comment by uklotzde** [comment]
I would also propose to rename the `URL` variants according to the Rust API naming conventions:

- `Url(UrlLinkFrame)`
- `UserUrl(ExtendedUrlFrame)`

This is an opportunity for doing it.

File: src/id3/v2/write/frame.rs [hunk 1]
```diff
@@ -26,24 +26,24 @@ where
...
 		| ("WFED" | "GRP1" | "MVNM" | "MVIN", FrameValue::Text { .. }) => Ok(()),
 		(id, FrameValue::Text { .. }) if id.starts_with('T') => Ok(()),
-		(id, FrameValue::URL(_)) if id.starts_with('W') => Ok(()),
+		(id, FrameValue::Url(_)) if id.starts_with('W') => Ok(()),
 		(id, frame_value) => Err(ID3v2Error::new(ID3v2ErrorKind::BadFrame(
 			id.to_string(),
 			match frame_value {
 				FrameValue::Comment(_) => "Comment",
-				FrameValue::UnSyncText(_) => "UnSyncText",
+				FrameValue::UnsynchronizedText(_) => "UnSyncText",
```
Comment (uklotzde) [inline src/id3/v2/write/frame.rs:42]: These strings need to be updated, too.

Comment (Serial-ATA) [reply src/id3/v2/write/frame.rs:42]: Good catch :smile:

**Review by uklotzde — APPROVED** [review]

**Pull request merged**